streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -824,6 +824,13 @@ object ZSink extends ZSinkPlatformSpecificConstructors {
       } yield push
     }
 
+  /**
+   * A sink that depends on another managed value
+   * `resource` will be finalized after the processing.
+   */
+  def managed[R, E, I, A, L, Z](resource: ZManaged[R, E, A])(fn: A => ZSink[R, E, I, L, Z]): ZSink[R, E, I, L, Z] =
+    ZSink(resource.fold[ZSink[R, E, I, L, Z]](err => ZSink.fail[E, I](err).dropLeftover, m => fn(m)).flatMap(_.push))
```

Comment:
Why are the leftovers dropped in the error case?

Reply:
`ZSink.fail` returns leftovers of type `I` and we require `L`.
I don't know how to properly treat allocation errors w/o loosing too much of ergonomics.

Reply:
Alternatively we could require ` ZManaged[R, Nothing, A]` but I expect most of the resources to be IO-related.

Reply:
Let’s require ZSink[R, E, I, I, Z] here. Dropping the leftovers is a bad choice.